YesterdayApi/Core/Email/Builder/EmailDirector.cs

```diff
@@ -0,0 +1,36 @@
+using MimeKit;
+using RazorHtmlEmails.RazorClassLib.Views;
+using YesterdayApi.Core.Email.Builder.Builders;
+using YesterdayApi.Core.Email.Builder.Renderer;
+using YesterdayApi.Core.Email.Builder.Types;
+
+namespace YesterdayApi.Core.Email.Builder
+{
+    public class EmailDirector : IEmailDirector
+    {
+        private EmailBuilder _emailBuilder;
+        private readonly IRazorViewToStringRenderer _razorViewToStringRenderer;
+
+        public EmailDirector(IRazorViewToStringRenderer razorViewToStringRenderer)
+        {
+            _razorViewToStringRenderer = razorViewToStringRenderer;
+        }
+
+        public MimeMessage PrepareEmail(IUserCredentials userCredentials, Type type)
+        {
+            PrepareBuilder(type);
+            _emailBuilder.ResetEmail();
+            _emailBuilder.AppendUserDetails(userCredentials);
+            _emailBuilder.PrepareBody();
+            return _emailBuilder.BuildEmail();
```

Comment:
Ahh i don't know if i would like to do this like 
var email = _emailBuilder.BuildEmail(); // would require less time to see whate the function returns
return email;

Reply:
@fabiankoziolek would you be able to tell what a MimeMessage is without clicking f12? I probably wouldn't, an email variable seems clearer to me. @rafalschmidt97 what are your thoughts on this.

Reply:
You're right, changed it